### DISCUSSION TITLE
`deleteUser()`: Add option to delete user-owned files

### DISCUSSION THREAD

**Discussion by imageck** [supabase/supabase#15424] [Feature Requests]
Currently, if a user owns any object in the storage, their account is not allowed to be deleted. The way to do it right now is by using triggers, cron jobs and such. This is not a trivial task, nor is it seamless.

Ideally, `deleteUser()` should accept an option akin to `shouldSoftDelete`, perhaps even named similarly, e.g. `shouldHardDelete`. From there on, there is a couple of options:

1. Start the deletion process asynchronously, then return the status.
2. Clear the `owner` column from `storage.objects`, marking those files for deletion at a later point, e.g. by a cron job.

**Comment by GaryAustin1** [comment]
I use this method to do what you describe with a trigger function and cron task...
https://github.com/GaryAustin1/supa-file-helper
